pytorch_lightning/trainer/training_tricks.py

```diff
@@ -45,12 +48,31 @@ def clip_gradients(self):
             for p in parameters:
                 p.grad.data.mul_(torch.where(clip_coef < 1, clip_coef, torch.tensor(1., device=device)))
 
-    def print_nan_gradients(self):
+    def print_nan_gradients(self) -> None:
         model = self.get_model()
         for param in model.parameters():
             if (param.grad is not None) and torch.isnan(param.grad.float()).any():
                 log.info(param, param.grad)
 
+    def detect_nan_tensors(self, loss: Tensor) -> None:
+        model = self.get_model()
+
+        # check if loss is nan
+        if not torch.isfinite(loss).all():
```

Comment:
so some can be Inf bot not all, right?

Reply:
the if condition is true when at least one element in the loss is either Nan, +inf, or -inf. Most of the time the loss is just one element (0-dim). But just in case that changes, I have added the .all()

Reply:
my bad overlooked `isfinate` as `infinite` lol